### DISCUSSION TITLE
32-bit arm build (armv7l)

### DISCUSSION THREAD

**Discussion by snpz** [openobserve/openobserve#2528] [Q&A]
Hi.

First of all, thanks for the great project! 

I'm experimenting with openobserve in my homelab to centralize logging and metrics. I'm having [rather old 32-bit hardware ](https://www.technexion.com/products/system-on-modules/evk/wandboard-imx6/) which is currently not officially supported.

I cloned the repo and was able to create a Dockerfile that uses cross-compilation on my windows pc in WSL to produce a docker image that is usable on my Wandboard.

I needed to do a little code change to make the build work, probably because the default value here is larger than a 32-bit int. (just guessing here, I'm not a rust developer)
``` git
snpz@DESKTOP:~/Dev/openobserve$ git diff src/config/src/config.rs
diff --git a/src/config/src/config.rs b/src/config/src/config.rs
index b511b28e..c260782f 100644
--- a/src/config/src/config.rs
+++ b/src/config/src/config.rs
@@ -945,8 +945,8 @@ fn check_disk_cache_config(cfg: &mut Config) -> Result<(), anyhow::Error> {
     cfg.limit.disk_free = disk_free as usize;
     if cfg.disk_cache.max_size == 0 {
         cfg.disk_cache.max_size = cfg.limit.disk_free / 2; // 50%
-        if cfg.disk_cache.max_size > 1024 * 1024 * 1024 * 100 {
-            cfg.disk_cache.max_size = 1024 * 1024 * 1024 * 100; // 100GB
+        if cfg.disk_cache.max_size > 1024 * 1024 * 512 {
+            cfg.disk_cache.max_size = 1024 * 1024 * 512; // 512MB
         }
     } else {
         cfg.disk_cache.max_size *= 1024 * 1024;
```

To get the collector installed I made another change in the [install.sh-script](https://github.com/openobserve/agents/blob/main/linux/install.sh) of the collector to support armv7 architecture.
Now metric collections works as expected.

I'm currently seeing some internal server errors with the [host-metrics dashboard](https://github.com/openobserve/dashboards), but that has probably nothing to do with the underlying architecture and I will retry again with the next release.

My question is: 
Is support for 32-bit armv7 something that could make it in the project or don't you want to support 32-bit at all in the future?

**Comment by hengfeiyang** [comment]
@snpz As you mentioned the code we can make some changes let you can build the docker image, but i don't think we will provider 32bit images. 

Please tell us if you got some error, we will help you on that.

**Reply by snpz** [reply]
Thanks for your swift reply! Eventually the time has finally come for me to switch my hardware in the near future. :)

I think the change in the agents repo would eventually make sense for other users as well because a lot of Raspberry Pis out there are still running a 32-bit OS which would make the otel-collector install.sh script work on these systems. If it is okay with you I will create a PR for this.

Regarding the change in the server I'm not sure. If you don't want to support 32-bit binaries would it make sense to make changes only relevant for those systems? Maybe it is better I create a fork to keep this out of the repository. What do you think?

**Reply by snpz** [reply]
@hengfeiyang 
I tested a bit more and although the 32-bit build is working on the surface I'm getting errors during some xhr requests in the browser like this one:
``` bash
openobserve  | thread 'actix-server worker 0' panicked at /rustc/2d7be73931e0978c8758a672cc7258b417a7e999/library/core/src/ops/function.rs:166:5:
openobserve  | called `Result::unwrap()` on an `Err` value: IpcError("Invalid uncompressed length: -202744274805063648")
```
The request was a POST to /web/logs?stream_type=logs&stream=journald&period=1d&refresh=0&type=stream_explorer&org_identifier=default and works without issues when openobserve is running via docker on my windows desktop using the same data-volume.

So there is probably more than one place that would need adjustment to make the server (I'm testing with 0.8.0-rc1) fully work on 32-bit and it is probably not worth the trouble to fix it, especially because I'm not able to contribute and 32-bit is dying.

Thank you for offering your help, but I will have to wait for a new Raspi 5 to become an openobserve user. :)
